pkg/service/targetdeploymentsvc/create.go

```diff
@@ -0,0 +1,57 @@
+package targetdeploymentsvc
+
+import (
+	"context"
+
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+)
+
+func (s *Service) CreateTargetGroupDeployment(ctx context.Context, req types.CreateTargetGroupDeploymentRequest) (*targetgroup.Deployment, error) {
+
+	// TODO: run pre-lim checks to ensure aws account/arn are valid
+	dbInput := targetgroup.Deployment{
+		ID:           req.Id,
+		FunctionARN:  req.FunctionArn,
+		Runtime:      req.Runtime,
+		AWSAccount:   req.AwsAccount,
+		Healthy:      false,
+		Diagnostics:  []targetgroup.Diagnostic{},
+		ActiveConfig: map[string]targetgroup.Config{},
+		Provider:     targetgroup.Provider{},
+	}
+
+	dbInput.Provider.Name = req.Provider.Name
```

Comment:
This will come from the healthcheck service, so it should be removed here. It should also be removed from the API request